scaling_performance/optimizing_storage.adoc

```diff
@@ -13,6 +13,81 @@ toc::[]
 
 == Optimizing Storage
 
+== General Storage Guidance
+
+The following is a matrix which details the storage backend types {product-title} can be used with implementation examples and vendors.
+
+=== Available Storage Options
+[format="csv",width="70%",cols="3"]
+[options="header]
+|=======================
+Type,Examples,Vendors
+hostPath,"hostPath pod mounted volumes.  A specific type of Block Storage.  File paths on the node mounted onto running pods.",Others
```

Comment:
@juanluisvaladas It can be one or the other.  https://kubernetes.io/docs/concepts/storage/volumes/#hostpath

@rootfs Current Hosted Metrics and Logging documentation are (intentionally?) vague here.  Some have implemented those with hostPath when faced with having a cluster that otherwise only has access to NFS storage.  And it is used at scale.  Those implementers have also created labeling and scheduler affinity/anti-affinity rules to support that use-case.  Part of why that was done (probably) was that there was no other obvious guidance.  I'm hope this PR rectifies that for some future implementers.